src/remote-config/remote-config-api-client.ts

```diff
@@ -174,11 +166,33 @@ export class RemoteConfigApiClient {
     }
     return this.sendPutRequest(template, ifMatch)
       .then((resp) => {
-        if (!validator.isNonEmptyString(resp.headers['etag'])) {
-          throw new FirebaseRemoteConfigError(
-            'invalid-argument',
-            'ETag header is not present in the server response.');
-        }
+        this.validateResponseEtag(resp.headers['etag']);
+        return {
```

Comment:
This can also go into a util right? Something like:

```
return this.createRemoteConfigTemplate(resp);
```

The util can do the validation and then extract the right properties from the response.

Reply:
Great point! Moved the creation to a new util. Thanks!